### PR TITLE
Reinstate the end method

### PR DESCRIPTION
**Pull request by KyeNormanGill**
**Please describe the changes this PR makes and why it should be merged:**
This PR re-adds the `end` method but named as `finish` for consistency with the finish event. This is to bring back the reasons *so i can get my playlists working again* omegaLUL

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/voice/dispatcher/StreamDispatcher.js [hunk 1]
```diff
@@ -142,6 +142,15 @@ class StreamDispatcher extends Writable {
     if (this._writeCallback) this._writeCallback();
   }
 
+  /**
+   * Stops the current stream permanently and emits a `finish` event.
+   * @param {string} reason The reason for ending this stream.
```
Comment (lloti) [inline src/client/voice/dispatcher/StreamDispatcher.js]: Shouldn't this reason be wrapped in brackets as its's optional?

**Comment by gmedeadlystrike** [comment]
I support this change. The removal of the ability to pass a reason when ending/destroying a dispatcher forced me to create some less then clean workarounds to allow for the .end event to understand why the dispatcher closed for post-processing

File: src/client/voice/dispatcher/StreamDispatcher.js (hunk 1 above)
Comment (KyeNormanGill) [reply src/client/voice/dispatcher/StreamDispatcher.js]: Good spot!

**Comment by appellation** [comment]
Considering that the stream itself will emit a `finish` event when ended, won't this change cause two finish events to be emitted with different parameters?  Considering that the underlying stream API does not support end reasons, I'm reticent to approve something that monkey-patches them in.  While I understand that end reasons are quite convenient, I personally believe that the concept encourages laziness; termination state should be managed by the audio player, not just the cleanup handler.

**Comment by GregoryGregoryGregory** [comment]
I support this change, the ability to pass a reason is a very useful feature.

**Comment by hydrabolt** [comment]
at the moment I'm against this PR, there's better ways to solve this than to implement a finish method.

**Comment by FireController1847** [comment]
I'm both ways. I can see where he would cause the need for this (*cough https://github.com/discordjs/discord.js/pull/2290 told you this was gonna happen*), but at the same time I see how this would cause double-events and there are much better ways to do this. -1

**Comment by KyeNormanGill** [comment]
I tested this and oddly enough the finish event only fired once on both ending naturally and using the method. Otherwise i wouldn't have PRed it.

**Comment by iCrawl** [comment]
This should be handled differently instead of just bringing back an old method for the sake of having it back.

**Pull request closed without merging**